backend/module/eCampApi/config/Rpc/auth.config.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+use eCampApi\RpcConfigFactory;
+use eCampApi\V1\Rpc\Auth\AuthController;
+use Laminas\Stdlib\ArrayUtils;
+
+$eCampAuth = RpcConfigFactory::forRoute('e-camp-api.rpc.auth')
+    ->setController(AuthController::class)
+    ->setRoute('/api/auth[/:action]')
+    ->addParameterDefault('action', 'index')
+    ->setAllowedHttpMethods('GET', 'POST')
+    ->setCollectionQueryWhiteList('callback')
+    ->build()
+;
+
+//only define collection query whitelist once, because it will me merged on controller basis.
+$googleAuth = RpcConfigFactory::forRoute('e-camp-api.rpc.auth.google')
+    ->setController(AuthController::class)
+    ->setRoute('/api/auth/google')
+    ->addParameterDefault('action', 'google')
+    ->setAllowedHttpMethods('GET', 'POST')
+    ->build()
+;
+
+//only define collection query whitelist once, because it will me merged on controller basis.
+$pbsMiDataAuth = RpcConfigFactory::forRoute('e-camp-api.rpc.auth.pbsmidata')
+    ->setController(AuthController::class)
+    ->setRoute('/api/auth/pbsmidata')
+    ->addParameterDefault('action', 'pbsmidata')
+    ->setAllowedHttpMethods('GET', 'POST')
+    ->build()
+;
+//use $eCampAuth as the last, that the route name for the controller
+//when setting the allowed HTTP methods is e-camp-api.rpc.auth
+$firstMerge = ArrayUtils::merge($pbsMiDataAuth, $googleAuth);
```

Comment:
How about you add a method `RpcConfigFactory#merge` so you could do
```php
$eCampAuth->merge($pbsMiDataAuth);
$eCampAuth->merge($googleAuth);
```

Reply:
i would rather not do merging myself, but leave that to Laminas\Stdlib\ArrayUtils.
I copied it from Laminas\Config\Factory::fromFiles and think we don't have to merge better than
the code using the config.
I hope we don't have to configure different HTTP Methods for the same Controller often in the future.
I would rather investigate further my theory, that you don't have to use the controller as key in the api-tools-rpc section.
If we can use the route name there, our lives get a lot easier.